### PR TITLE
MPD: version bump, change to meson based build system

### PR DESCRIPTION
**Pull request by extrowerk**
No Haiku native audio output, but openAL works nicely.
All test passes.
Buildsystem needs more love, but it is ok for now.

### CODE REVIEW

**Comment by korli** [comment]
Unless you tested x86 secondary architecture, I would leave the previous version enabled for it and disabled for the new version.

**Comment by waddlesplash** [comment]
@Begasus, can you test on x86?

File: media-sound/mpd/mpd-0.21.3.recipe [hunk 1]
```diff
@@ -139,34 +145,48 @@ BUILD_REQUIRES="
 	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:autoconf
+	setuptools_python3
```
[the remark this answers is not in the document]
Comment (waddlesplash) [reply media-sound/mpd/mpd-0.21.3.recipe]: No, it's a host tool, it belongs in PREREQUIRES.

File: media-sound/mpd/mpd-0.21.3.recipe [hunk 2]
```diff
@@ -139,34 +145,48 @@ BUILD_REQUIRES="
...
 {
-	make install
+	cd haiku_build
+	ninja install
+
+	# Should be fixed in the build-system
+	mkdir -p $binDir
+	cp mpd $binDir
+	rc -o mpd.rsrc ../src/haiku/mpd.rdef
+	xres -o $binDir/mpd -- mpd.rsrc
```
[the remark this answers is not in the document]
Comment (Begasus) [reply media-sound/mpd/mpd-0.21.3.recipe]: This did the trick into building it for _x86 on gcc2h
```diff
-	mkdir -p $binDir
-	cp mpd $binDir
-	rc -o mpd.rsrc ../src/haiku/mpd.rdef
-	xres -o $binDir/mpd -- mpd.rsrc
+	cp mpd $commandBinDir
+	rc -o mpd.rsrc ../src/haiku/mpd.rdef
+	xres -o $commandBinDir/mpd -- mpd.rsrc
+	chmod +x $commandBinDir/mpd
```

File: media-sound/mpd/mpd-0.21.8.recipe [hunk 3]
```diff
@@ -139,34 +144,49 @@ BUILD_REQUIRES="
 	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:autoconf
+	setuptools_python3
 #	cmd:avahi_client$secondaryArchSuffix
-	cmd:awk
-	cmd:find
+	cmd:cmake
```
Comment (korli) [inline media-sound/mpd/mpd-0.21.8.recipe]: are you sure about cmake

Comment (extrowerk) [reply media-sound/mpd/mpd-0.21.8.recipe]: It definetely searching for it. I was also surprised, but maybe it using cmake in pkg-config mode (it is possible, btw) to search for its deps.

**Comment by extrowerk** [comment]
AFAIK the meson build system doesn't know how to build the haiku output module, so it isn't supported, only openal output usable.

File: media-sound/mpd/mpd-0.21.8.recipe (hunk 3 above)
Comment (korli) [reply media-sound/mpd/mpd-0.21.8.recipe]: or there are third party deps which could require cmake, dunno

**Comment by korli** [comment]
well i see it added:
https://github.com/MusicPlayerDaemon/MPD/blob/master/src/output/plugins/meson.build#L29

Maybe haiku isn't correctly detected:
https://github.com/MusicPlayerDaemon/MPD/blob/master/meson.build#L102

**Comment by extrowerk** [comment]
Afaik meson uses Python, and Haiku declared as "haiku1" in python. I will try to change that line, and we will see, what happens.

**Comment by extrowerk** [comment]
Just changed the meson script to print the osname and it prints just "haiku", so the tests should be ok, but still doesn't builds the haiku plugins. Any idea, whats wrong?

**Pull request merged**